tutorial/markdown/kotlin/ktor.md

```diff
@@ -102,195 +95,356 @@ fun createScope(bucket: Bucket, configuration: CouchbaseConfiguration): Scope {
-The search is implemented under the "/profile/search" url in the last handler function inside `ProfileRoutes.kt` file.
+This endpoint fetches the airports that can be reached directly from the specified source airport code. This also uses a SQL++ query to fetch the results similar to the List Airport endpoint.
 
-Search handfler function processes GET requests and uses `ProfileSearchRequest` data class to accept profile search string and optional pagination parameters.
-The actual search is again done by a repository function:
+Let us look at the query used here:
 
+```sql
+SELECT distinct (route.destinationairport)
+FROM airport as airport
+JOIN route as route on route.sourceairport = airport.faa
+WHERE airport.faa = $airport and route.stops = 0
+ORDER BY route.destinationairport
+LIMIT $limit
+OFFSET $offset
 ```
-fun search(text: String, skip: Int = 0, limit: Int = 10): List<Profile> {
-    return runBlocking(databaseContext) {
-        val lowerText = text.lowercase()
-        val qryString = "SELECT p.* FROM `profile` p " +
-                "WHERE lower(p.firstName) LIKE '%$lowerText%' " +
-                "OR lower(p.lastName) LIKE '%$lowerText%' " +
-                "LIMIT $limit OFFSET $skip"
-        val result = collection.scope.query(qryString, readonly = true).execute()
-        result.rows.map {
-            it.contentAs<Profile>()
-        }.toList()
-    }
-}
-```
-> _from [`src/main/kotlin/com/couchbase/kotlin/quickstart/ProfileRepository.kt`](https://github.com/couchbase-examples/kotlin-quickstart/blob/main/src/main/kotlin/com/couchbase/kotlin/quickstart/ProfileRepository.kt)_
 
-We use `LIKE` SQL++ operator and lower case strings to perform case-insensitive lookup for all profile objects that contain the search string either in first or last name.
+We are fetching the direct connections by joining the airport collection with the route collection and filtering based on the source airport specified by the user and by routes with no stops.
 
-### Running The Tests and using the API
+## Running The Tests
+
+We have defined integration tests using the [JUnit](https://kotlinlang.org/docs/jvm-test-using-junit.html) package for all the API end points. The integration tests use the same database configuration as the application. For the integration tests, we perform the operation using the API and confirm the results by checking the documents in the database. For example, to check the creation of the document by the API, we would call the API to create the document and then read the same document from the database and compare them. After the tests, the documents are cleaned up by calling the DELETE endpoint
 
 To run the standard integration tests, use the following commands:
 
-```shell
+```sh
 ./gradlew test
 ```
 
-And, to run the application, use `./gradlew run`, which should create an HTTP server with OpenAPI UI at [http://localhost:8080](http://localhost:8080)
+## Appendix
+
+### Data Model
```

Comment:
You don't need the Data Model here as it is already described before.

Reply:
corrected it.